specification/paths/CarrierContracts-carrier_contract_id.json

```diff
@@ -39,5 +39,97 @@
         }
       }
     }
+  },
+  "patch": {
+    "tags": [
+      "CarrierContracts"
+    ],
+    "summary": "Update an carrier contract by id.",
+    "description": "This endpoint updates the carrier contract with the supplied id.",
+    "parameters": [
+      {
+        "$ref": "#/components/parameters/path-carrier_contract_id"
+      }
+    ],
+    "requestBody": {
+      "description": "The carrier contract object with attributes to be updated.",
+      "required": true,
+      "content": {
+        "application/vnd.api+json": {
+          "schema": {
+            "type": "object",
+            "required": [
+              "data"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "data": {
+                "allOf": [
+                  {
+                    "$ref": "#/components/schemas/CarrierContract"
+                  },
+                  {
+                    "required": [
+                      "id"
+                    ],
+                    "properties": {
+                      "attributes": {
+                        "properties": {
+                          "credentials": {
+                            "type": "string",
+                            "example": "{\"api_user\":\"your_api_user\",\"api_password\":\"your_password\"}"
```

Comment:
This example looks weird. It seems that escaping double quotes isn't possible in our spec. Have you considered just using type "object" and then in the API converting the object into json string? That also looks nicer for the person making the request. Instead of writing one long string of JSON.